src/java.base/share/classes/sun/nio/ch/Net.java

```diff
@@ -679,6 +679,10 @@ static boolean pollConnectNow(FileDescriptor fd) throws IOException {
      */
     static native int sendOOB(FileDescriptor fd, byte data) throws IOException;
 
+    /**
+     * Read and discard urgent data (MSG_OOB) on the socket.
+     */
+    static native boolean discardOOB(FileDescriptor fd) throws IOException;
```

Comment:
I see a native implementation of this method on Windows only, but it is in the shared `Net` class. This method is not called on non-Windows, but it it were there would be a problems ;-).  Was the intention to add other platform specific implementations, or to confine it to Windows only?

Reply:
It often happens in this area that Net and IOUtil define methods that aren't implemented on all platforms. In this case, discardOOB is generally useful and we can add it on other platforms when needed. I didn't add it now because it would not be used or tested.

Reply:
This attracted my attention because it is a new native method. I don't see any other native methods unimplemented here, but I only skimmed over the code. Seems very fragile, since it will (may) throw a linkage error if called on non-Windows platforms - without any help from the compiler, we are dependent on good tests to catch such. If this is an established pattern, then ok, but I would like to revisit it separately to this PR.

Reply:
IOUtil.makePipe is another example. It has been there since JDK 1.4 but doesn't have an implementation on Windows. The Net.localXXX methods didn't have implementations on all platforms for a long time too. So yes, if someone where to write some platform specific code in sun.nio.ch and use one of the unimplemented methods then it would be an error at runtime.